src/Commands/new.js

```diff
@@ -2,88 +2,238 @@
+    this.applicationName = args.name
+    this.applicationPath = path.join(process.cwd(), args.name)
+    this.mustUse = null
+
+    if (options.npm) {
+      this.mustUse = 'npm'
+    } else if (options.yarn) {
+      this.mustUse = 'yarn'
+    }
+
+    this._dumpAsciiLogo()
+    this._checkRequirements()
+    yield this._verifyApplicationDoesntExist()
+    yield this._cloneRepository()
+
+    if (options['skip-install'] === null) {
+      yield this._installDependencies()
+    }
+
+    yield this._cleanProjectDirectory()
+    yield this._copyEnvironmentFile()
+
+    if (options['skip-install'] === null) {
+      yield this._generateSecureKey()
+    }
+
+    this.log()
+    this.success(`${this.icon('success')} Your application is ready!`)
+    this.log()
+
+    this.info(`${this.icon('info')} Follow below instructions to get started`)
+    this.log(`$ cd ${this.colors.magenta.bold(args.name)}`)
+    this.log(`$ npm run serve:dev`)
+    this.log()
+  }
 
-const repo = 'https://github.com/adonisjs/adonis-app.git'
-let branch = 'master'
+  /**
+   * Verify that the application folder doesn't already exists.
+   *
+   * @private
+   * @method _verifyApplicationDoesntExist
+   * @return {void}
+   */
+  * _verifyApplicationDoesntExist () {
+    try {
+      yield pify(fs.access)(this.applicationPath, fs.constants.F_OK)
+      this.error(`${this.icon('error')} The directory "${this.applicationName}" already exists!`)
+      process.exit(0)
+    } catch (e) {}
+  }
 
-module.exports = function (argv) {
   /**
-   * switch to develop branch if --dev option is passed
-   * in commandline
+   * Clone the AdonisJs blueprint repository with
+   * the given branch.
+   *
+   * @private
+   * @method _cloneRepository
+   * @return {void}
    */
-  if (argv.dev) {
-    branch = 'develop'
+  * _cloneRepository () {
+    const branch = this.colors.magenta.bold(this.branch)
+    const blueprint = this.colors.magenta.bold(this.blueprint)
+
+    this._startSpinner(
+      this.colors.blue(`Cloning ${branch} version of ${blueprint} blueprint`)
+    )
+
+    try {
+      yield pify(exec)(`git clone -b ${this.branch} --single-branch https://github.com/${this.blueprint}.git ${this.applicationPath}`)
+      this._stopSpinner()
+      this.log()
+      this.completed('clone', 'Repository cloned')
+    } catch (e) {
+      this.log()
+      this.error(`${this.icon('error')} An error occured while trying to clone ${branch} version of ${blueprint} blueprint.`)
+      process.exit(0)
+    }
+  }
+
+  /**
+   * Install all dependencies of the application.
+   *
+   * @private
+   * @method _installDependencies
+   * @return {void}
+   */
+  * _installDependencies () {
+    let tool = 'npm'
+
+    if (!this.mustUse && (yield this._hasYarnInstalled())) {
+      this.info(`${this.icon('info')} Yarn has been detected in your system!`)
+      if (yield this.confirm('Do you want to use it instead of npm?', false).print()) {
+        tool = 'yarn'
+      }
+    } else {
```

Comment:
This should be else if, If I do not have yarn installed and I neither pass `--npm` or `--yarn` flag, It will say.. 

> Installing dependencies using null

Since `mustUse` is null by default. So it should be

```
else if (this.mustUse) {
   tool = this.mustUser
}
```